### ISSUE TITLE
Parameters management division between owner and committee

### ISSUE THREAD

**Issue by ben-kaufman** [hats-finance/hats-contracts#280]
Some parameters like setBountySplit are managed by the owner, while others like setMaxBounty are managed by the committee. It's unclear why some parameters are settable by committee and others by the owner? Seems quite arbitrary...

**Comment by jellegerbrandy** [comment]
So these methods, right?

Committee:
- `setPendingMaxBounty` and `setMaxBounty`

Vault Owner:
- `setBountySplit` - split between hacker, committe, governance, etc
- `setVestingParams`

I guess governance should at least be able to set the part of the bounty that should go to governance (otherwise the committee will set it to 0). The comittee controls the size of the bounty. 

It does not look that arbitratry to me.

**Comment by ben-kaufman** [comment]
Ok, but also for example pause deposits is controller by the owner, and the descriptionHash too.

**Comment by ben-kaufman** [comment]
I think we should probably have the pause deposits controller by the committee, and seperated to its own function. And have the _visible and maybe also _descriptionHash controlled by the owner, and also be in the registry itself, so it's easier for governance to manage (instead of being in each vault contract as now in #276

**Comment by ben-kaufman** [comment]
Also setVestingParams of individual vaults is not related to the governance, why is it set by governance and not committee?

And for the bounty split, why is the setBountySplit part of the governance not the same for all vaults?

**Comment by ben-kaufman** [comment]
Also setRewardController, I see no reason it should be the owner and not the committee

**Comment by jellegerbrandy** [comment]
I updated the list of functions and who can call them here: https://github.com/hats-finance/hats-contracts/blob/update-docs/docs/roles.md 

I guess we need some idea of the what the different roles are supposed to do to be able to answer your questions:

For now (i.e. in v1), the approach (it seems to me) is that the committee is reponsible for the bounties, and all other things are managed by the owner (i.e. hats governance). 

We now have two ower roles (registry and vault owner) so perhaps we should think about dividing the responsabilities between these two.. so we would get vault-specific parameters managed by vault owner, protocol-level stuff managed by registry owner.

So:
1. Committee: submits claims
2. Vault owner: manages vault-specific settings
3. Registry owner: manages global settings

We are not so far off from that, I think..

Regarding yr specific comments:
- +1 for moving the visibility and pausilbity functions to the registry
- descriptionHash - this defines the bug descriptions etc, and to me it seems it should be set by the committee (@shayzluf wdyt?)
- setVestingParams is about the vesting of HATs tokens. It is I think improtant that hats governance stays in control of that
- setBountySplit settable per vault: idk but you can imagine that a committee may agree not to be paid
- setRewardController: I think hats wants to stay in control of the rewards program, so this should be incontrol of governance

**Comment by jellegerbrandy** [comment]
Of course, there is also #277 to consider (i.e. we may just have one single owner variable for registry and all vaults)

**Comment by shayzluf** [comment]
> I updated the list of functions and who can call them here: https://github.com/hats-finance/hats-contracts/blob/update-docs/docs/roles.md 
> 
> I guess we need some idea of the what the different roles are supposed to do to be able to answer your questions:
> 
> For now (i.e. in v1), the approach (it seems to me) is that the committee is reponsible for the bounties, and all other things are managed by the owner (i.e. hats governance). 
> 
> We now have two ower roles (registry and vault owner) so perhaps we should think about dividing the responsabilities between these two.. so we would get vault-specific parameters managed by vault owner, protocol-level stuff managed by registry owner.
> 
> So:
> 1. Committee: submits claims
> 2. Vault owner: manages vault-specific settings
> 3. Registry owner: manages global settings
> 
> We are not so far off from that, I think..
> 
> Regarding yr specific comments:
> - +1 for moving the visibility and pausilbity functions to the registry
> - descriptionHash - this defines the bug descriptions etc, and to me it seems it should be set by the committee (@shayzluf wdyt?)
> - setVestingParams is about the vesting of HATs tokens. It is I think improtant that hats governance stays in control of that
> - setBountySplit settable per vault: idk but you can imagine that a committee may agree not to be paid
> - setRewardController: I think hats wants to stay in control of the rewards program, so this should be incontrol of governance
> 
> 

Main things to take into account:
Governance reward + part of the reward that is exchanged to hats and its timing should be set by governance .(no incentive for the committee to do it at all)

Committee reward should have an upper bound

The description hash and rewards can be set by committee but should have timelock to avoid submission with one set of rules to get a different reward? (Maybe there is no need for a timelock because it could be proved onchain that the committee changed it after the submission)

Who is the vault owner?

**Comment by ben-kaufman** [comment]
> setVestingParams is about the vesting of HATs tokens. It is I think improtant that hats governance stays in control of that

No you're confusing with setHATVestingParams. The normal setVestingParams controls vesting of the normal vault token so I think it should be moved to control of the committee

**Comment by lirona** [comment]
- for separating the registry and vaults owners, i'm a bit confused what will be the vault owner if not hats governance.
- the visibility and pausability functions are vault specific so i don't see a reason to have it in the registry and give the specific vault's address, but that depends on the previous point
- i think pause deposits should be able to be done by the committee and governance since it's for emergencies, visibility makes sense it's only governance, and description should be committee (now all of them can only be changed by governance)
- agree that `setVestingParams` should be set by committee and not gov, and `setHatVestingParams` stays under gov.
- agree that bounty split part for gov should be the same in all vaults, and also should only be set by gov (and then if changed by gov we need to change all the vaults' bounty splits accordingly) and rest of bounty split set by committee
- not sure about setRewardController, because anyway it should always be owned by governance (transfers ownership on initialization)
- agree that committee reward should have an upper bound, and also gov reward

**Comment by ben-kaufman** [comment]
> agree that bounty split part for gov should be the same in all vaults, and also should only be set by gov (and then if changed by gov we need to change all the vaults' bounty splits accordingly) and rest of bounty split set by committee
I agree. So should we split that setter? Have the governance part in the registry and the rest in each vault managed by the committee? @jellegerbrandy @shayzluf 

> i think pause deposits should be able to be done by the committee and governance since it's for emergencies, visibility makes sense it's only governance, and description should be committee (now all of them can only be changed by governance)

I agree it should be by both. @jellegerbrandy @shayzluf what do you think?

> agree that committee reward should have an upper bound, and also gov reward
What should these be then?

**Comment by lirona** [comment]
> > agree that bounty split part for gov should be the same in all vaults, and also should only be set by gov (and then if changed by gov we need to change all the vaults' bounty splits accordingly) and rest of bounty split set by committee
> 
> I agree. So should we split that setter? Have the governance part in the registry and the rest in each vault managed by the committee? @jellegerbrandy @shayzluf
yes, one function for the committee to set:
        hacker;
        hackerVested;
        committee;
and one function for the gov to set for all vaults:
        swapAndBurn;
        hackerHatVested;
        governanceHat;
@jellegerbrandy @shayzluf ?
> 
> > i think pause deposits should be able to be done by the committee and governance since it's for emergencies, visibility makes sense it's only governance, and description should be committee (now all of them can only be changed by governance)
> 
> I agree it should be by both. @jellegerbrandy @shayzluf what do you think?
> 
> > agree that committee reward should have an upper bound, and also gov reward
> 
> What should these be then?
> 
> > agree that setBountySplit should have a timelock
> 
> Well if we split it, then the part managed by the committee will not be under the timelock controller anymore, but we can add a delay like the maxBounty delay (maybe use the same delay?)
think so

**Comment by jellegerbrandy** [comment]
I am not sure at all about these changes, or why this is an issue now. 

In v1, basically, the policy is: the committee evaluates and submits the bounties, governance manages all parameters and basically everything else with a time lock. 

This issue proposes to change that, and give more responsibility to the committee. Why do you want to do this now and not one year ago? 

Also, what is your reasoning here? Why would, for example, the committee control the tokenvesting time? I am on the d-protocol committee. I know nothing at all about their token, their release schedule, if they at the moment would be happy with  some more liquidity or instead they prefer to have long locking times.  I am on the committee bc presumably I can evaulate how serious a bug is, not bc I am an expert in the tokenomics of their token.

**Comment by jellegerbrandy** [comment]
Also description hash.  This is content that may be shown on the HATs finance site. Is it ok to give up control over that?

**Comment by jellegerbrandy** [comment]
Anyway, my main point is: nobody is unhappy with the status quo afaik. Why change it?

**Comment by lirona** [comment]
guess we're confusing the committee with the project leadership.. for now everything that should be done by the project's leadership is done through hats gov, and maybe that's another role to define later.. but anyway i accept what you're saying, but still think that pausing deposits should be also done by committee, and that the committee and gov shares of bounties should be limited

**Comment by ben-kaufman** [comment]
So the main reason for all this is that it just doesn't seem to make sense to me that the hats governance needs to manage for all vaults parameters that depend a lot on the specifics of each individual project. I assumed the committee is basically the project leadership. But if not, then maybe we should have another role that is the project leadership so that projects can manage these types of configurations independently. This I think will also improve decentralization.

But if we want to keep everything under management of the hats governance we can do that too.
